### PR TITLE
Added grpc_alarm (renamed previous one to grpc_timer)

### PR DESCRIPTION
**Pull request by dgquintas**

### CODE REVIEW

File: src/core/surface/completion_queue.h [hunk 1]
```diff
@@ -51,6 +52,16 @@ typedef struct grpc_cq_completion {
   gpr_uintptr next;
 } grpc_cq_completion;
 
+/** An alarm associated with a completion queue. */
```
Comment (ctiller) [inline src/core/surface/completion_queue.h]: This needs to be a public API (in grpc.h).

Would prefer grpc_alarm for the type (yes I realize this needs a global search/replace for the existing type).

API looks right, though exec_ctx args will need to be dropped when going public - create the exec_ctx in her function instead - and api names will need to be adjusted to the new type name.

**Comment by ctiller** [comment]
Let's break the implementation and test into separate files from completion_queue. It's really a separate kind of object.

File: src/core/surface/completion_queue.h (hunk 1 above)
Comment (dgquintas) [inline src/core/surface/completion_queue.h]: If this new cq alarms take the grpc_alarm name, what should the current grpc_alarm from alarm.h be called?

Comment (ctiller) [inline src/core/surface/completion_queue.h]: Not sure... Maybe timer?

Comment (dgquintas) [inline src/core/surface/completion_queue.h]: maybe grpc_iomgr_alarm?

**Comment by dgquintas** [comment]
Done. PTAL.

**Comment by dgquintas** [comment]
Added C++ wrapper and its tests.

File: third_party/protobuf [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 8fce8933649ce09c1661ff2b5b7f6eb79badd251
```
Comment (ctiller) [inline third_party/protobuf]: no no

Comment (dgquintas) [inline third_party/protobuf]: sorry, fixed.

**Comment by ctiller** [comment]
Looks good: will merge when it's updated and had a Jenkins run.

**Pull request merged**